nifi-commons/nifi-security-utils/src/main/java/org/apache/nifi/security/util/CertificateUtils.java

```diff
@@ -469,6 +472,14 @@ public static X509Certificate generateSelfSignedX509Certificate(KeyPair keyPair,
             // (2) extendedKeyUsage extension
             certBuilder.addExtension(Extension.extendedKeyUsage, false, new ExtendedKeyUsage(new KeyPurposeId[]{KeyPurposeId.id_kp_clientAuth, KeyPurposeId.id_kp_serverAuth}));
 
+            // (3) subjectAlternativeName extension. Include CN as a SAN entry.
+            try {
+                final String cn = IETFUtils.valueToString(new X500Name(dn).getRDNs(BCStyle.CN)[0].getFirst().getValue());
+                certBuilder.addExtension(Extension.subjectAlternativeName, false, new GeneralNames(new GeneralName(GeneralName.dNSName, cn)));
```

Comment:
Most uses of common name use it as a DNS name, but some test classes use a generic description instead of a DNS name for the common name field.  Although it may not cover every use case, introducing a simple regular expression pattern to check for what appears to be a valid hostname could be used to determine whether or not to add the SAN.  The pattern should support unqualified hostnames, like `localhost` as well as qualified names like `localhost.localdomain`.

Reply:
How about *.domain.tld ? or *.sub.domain.tld?

Reply:
After evaluating the valid characters for a DNS name, it looks like it would be difficult to incorporate an appropriate regular expression pattern to cover all cases, so it is probably best to just introduce a check for a string that is not blank.